packages/shared/lib/core/network/stores/selected-connected-chain-index.store.ts

```diff
@@ -0,0 +1,24 @@
+import { activeProfile } from '@core/profile'
+import { Readable, Writable, derived, writable } from 'svelte/store'
+import { buildChainFromNetwork } from '../utils'
+import { ConnectedChain } from '../interfaces'
+import { NetworkHealth } from '../enums'
+
+export const selectedConnectedChainIndex: Writable<number> = writable(undefined)
+
+export const selectedConnectedChain: Readable<ConnectedChain> = derived(
```

Comment:
Not requesting a change, but just FYI - these will (I think) change slightly once we add the `IChain` class impls, but yeah not entirely sure just yet